src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -829,7 +838,8 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
 
                 dbResultSetRow = dbResultSet is not null ?
                     (dbResultSet.Rows.FirstOrDefault() ?? new DbResultSetRow()) : null;
-                if (dbResultSetRow is not null && dbResultSetRow.Columns.Count == 0)
+
+                if (dbResultSetRow is not null && dbResultSetRow.Columns.Count == 0 && dbResultSet!.ResultProperties.TryGetValue("RecordsAffected", out object? recordsAffected) && (int)recordsAffected <= 0)
```

Comment:
Looks like you're not guaranteed to have `recordsAffected` be non-null. So wouldn't the following && condition possibly fail in casting when recordsAffected is null?

Reply:
Based on my understanding of the records affected field: https://learn.microsoft.com/en-us/dotnet/api/system.data.common.dbdatareader.recordsaffected?view=net-8.0. It says that this field must be implemented on dbdatareader closed. Even for select statements it returns a -1 but is a valid value. In existing DAB code today, I see similar logic used in the delete operation: 
<img width="576" alt="image" src="https://github.com/Azure/data-api-builder/assets/124841904/dbdb0159-3b8c-4e63-b06a-da3280762c9b">